src/main/java/io/stargate/sgv2/jsonapi/service/cqldriver/executor/TableSchemaObject.java

```diff
@@ -115,4 +127,127 @@ public static TableSchemaObject from(TableMetadata tableMetadata, ObjectMapper o
     }
     return new TableSchemaObject(tableMetadata, vectorConfig);
   }
+
+  /**
+   * Convert table schema object to table response which is returned as response for `listTables`
+   *
+   * @return
+   */
+  public TableResponse toTableResponse() {
+    String tableName = removeQuotes(tableMetadata().getName());
+    HashMap<String, ColumnType> columnsDefinition = new HashMap<>();
+    for (Map.Entry<CqlIdentifier, ColumnMetadata> column :
+        tableMetadata().getColumns().entrySet()) {
+      ColumnType type = getColumnType(column.getKey().asInternal(), column.getValue());
+      columnsDefinition.put(removeQuotes(column.getKey()), type);
+    }
+
+    final List<String> partitionBy =
+        tableMetadata().getPartitionKey().stream()
+            .map(column -> removeQuotes(column.getName()))
+            .collect(Collectors.toList());
+    final List<PrimaryKey.OrderingKey> partitionSort =
+        tableMetadata().getClusteringColumns().entrySet().stream()
+            .map(
+                entry ->
+                    new PrimaryKey.OrderingKey(
+                        removeQuotes(entry.getKey().getName()),
+                        entry.getValue() == ClusteringOrder.ASC
+                            ? PrimaryKey.OrderingKey.Order.ASC
+                            : PrimaryKey.OrderingKey.Order.DESC))
+            .collect(Collectors.toList());
+    PrimaryKey primaryKey =
+        new PrimaryKey(
+            partitionBy.toArray(new String[0]),
+            partitionSort.toArray(new PrimaryKey.OrderingKey[0]));
+    return new TableResponse(
+        tableName, new TableResponse.TableDefinition(columnsDefinition, primaryKey));
+  }
+
+  // Need to handle frozen types
+  private ColumnType getColumnType(String columnName, ColumnMetadata columnMetadata) {
+    if (columnMetadata.getType() instanceof VectorType vt) {
+      // Schema will always have VectorConfig for vector type
+      VectorConfig.ColumnVectorDefinition columnVectorDefinition =
+          vectorConfig.columnVectorDefinitions().stream()
+              .filter(vc -> vc.fieldName().equals(columnName))
+              .findFirst()
+              .get();
+      VectorizeConfig vectorizeConfig =
+          columnVectorDefinition.vectorizeConfig() == null
+              ? null
+              : new VectorizeConfig(
+                  columnVectorDefinition.vectorizeConfig().provider(),
+                  columnVectorDefinition.vectorizeConfig().modelName(),
+                  columnVectorDefinition.vectorizeConfig().authentication(),
+                  columnVectorDefinition.vectorizeConfig().parameters());
+      return new ComplexTypes.VectorType(PrimitiveTypes.FLOAT, vt.getDimensions(), vectorizeConfig);
```

Comment:
Should we do any verification wrt vector being of "Float" type? I know we currently only support that, but is there any benefit from future-proofing?

Reply:
I left this to have a format hardcoded since cql driver has it. If and when cassandra adds support for other type, We can enable it with minimal changes and exposing it in api request.